src/main/java/org/terasology/climateConditions/VisibleBreathingSystem.java

```diff
@@ -0,0 +1,68 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.climateConditions;
+
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.location.Location;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.particles.components.generators.VelocityRangeGeneratorComponent;
+import org.terasology.registry.In;
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class VisibleBreathingSystem extends BaseComponentSystem {
+    public static final String VISIBLE_BREATH_ACTION_ID = "Visible Breath";
+
+    @In
+    private EntityManager entityManager;
+
+    @ReceiveEvent(components = {HypothermiaComponent.class})
+    public void onPeriodicBreath(PeriodicActionTriggeredEvent event, EntityRef player) {
+        if (event.getActionId().equals(VISIBLE_BREATH_ACTION_ID)) {
+            updateVisibleBreathEffect(player);
+        }
+    }
+
+    private void updateVisibleBreathEffect(EntityRef player) {
+        EntityRef particleEntity = entityManager.create("climateConditions:VisibleBreathEffect");
+        LocationComponent targetLoc = player.getComponent(LocationComponent.class);
+        LocationComponent childLoc = particleEntity.getComponent(LocationComponent.class);
+        childLoc.setWorldPosition(targetLoc.getWorldPosition());
+        Location.attachChild(player, particleEntity);
+        particleEntity.setOwner(player);
+        Vector3f direction = targetLoc.getLocalDirection();
+        direction.normalize();
+        //NOTE: initializing velocity using the constructor wasn't working hence the long method.
+        //particleEntity.addComponent(new VelocityRangeGeneratorComponent(direction,direction.scale(2)));
+        VelocityRangeGeneratorComponent velocity = particleEntity.getComponent(VelocityRangeGeneratorComponent.class);
+        direction.scale(0.5f);
+        direction.addY(0.5f);
+        velocity.minVelocity = direction;
+        direction.scale(1.5f);
+        velocity.maxVelocity = direction;
+        particleEntity.addOrSaveComponent(velocity);
```

Comment:
See the comment below for a more detailed explanation, but this can also use `update` or `upsert`. 
As it is currently written here, it will fail if the component is not present on the entity (NPE when accessing `velocity`). 
The main difference between `update` and `upsert` is that the former will only "execute" if the entity already has the component. If not, nothing will happen. 

Without suggesting anything here, it's up to you to decide which of both is the right one to use here. I'll give an example for `update` as that is closer to the current code:

```suggestion
        particleEntity.update(VelocityRangeGeneratorComponent.class, velocity => {
	        direction.scale(0.5f);
	        direction.addY(0.5f);
	        velocity.minVelocity = direction;
	        direction.scale(1.5f);
	        velocity.maxVelocity = direction;
        });
```